### DISCUSSION TITLE
数据集标签及训练问题

### DISCUSSION THREAD

**Discussion by IceHowe** [PaddlePaddle/PaddleOCR#13629] [Q&A]
你好，我这边用PP-OCRv4，第一次训练手写数据集，用上面那张图类型的标签是可以训练的，第二次训练印章的OCR，他的数据标签格式是第二张图的格式，我看官网https://paddlepaddle.github.io/PaddleOCR/applications/%E5%8D%B0%E7%AB%A0%E5%BC%AF%E6%9B%B2%E6%96%87%E5%AD%97%E8%AF%86%E5%88%AB.html#31 写的应该是支持的，可我这俩用这种方式无法正常运行，也不报错，会卡在一个地方，也就是第三张图展示的样子
![image](https://github.com/user-attachments/assets/80d7d4b1-c80e-462d-a7d2-130d38b03f68)
![image](https://github.com/user-attachments/assets/8b5d78ad-5147-4d0c-a553-6d6f36f19570)
![image](https://github.com/user-attachments/assets/2aa84231-ca96-462e-866c-68b7f810ad06)

**Comment by SWHL** [comment]
建议把num workers改为0，batch size =1 再跑，应该就能有具体报错信息了。

**Reply by IceHowe** [reply]
改了后也是不报错，感觉是数据集格式问题，问题图二的数据集标签格式有问题吗？难道只支持问题图一的格式吗？可OCR文字检测也是这种情况，不只是OCR文字识别这样
![image](https://github.com/user-attachments/assets/4de1e9e1-d269-4cfa-8eb5-ad2d90d6ebe4)
![image](https://github.com/user-attachments/assets/314af01f-66a8-46e5-b246-e53c4dbaddf6)

**Reply by IceHowe** [reply]
哦哦报错了：
![image](https://github.com/user-attachments/assets/3e1d50ec-b626-48e3-9f71-2c64d50dca5a)
![image](https://github.com/user-attachments/assets/517eb905-2ab7-4dd1-87a4-a0ac222942ca)
![image](https://github.com/user-attachments/assets/7431d399-9d3c-429c-89ed-19fbe54cd560)
![image](https://github.com/user-attachments/assets/9690f89b-fc7a-4b44-ad94-d0eccfd76ec6)

**Reply by IceHowe** [reply]
> 建议把num workers改为0，batch size =1 再跑，应该就能有具体报错信息了。

谢谢

**Comment by UserWangZz** [comment]
识别好像不支持图2这种格式

**Reply by IceHowe** [reply]
只能用图一的那种格式吗？我看到官网的PPOCR印章识别的应用示例里用的是这种格式，还是说已经改了？针对这种情况有什么好的建议吗？谢谢
![image](https://github.com/user-attachments/assets/8c419b9e-bc3b-4209-8431-2c608d5f0736)

**Reply by IceHowe** [reply]
印章识别不是包含两部分嘛，第一部分是目标检测，第二部分是OCR（OCR文字检测和OCR文字识别），对于第二部分OCR文字检测和识别都不会有这种情况

**Reply by GreatV** [reply]
用的是pgnet模型吗

**Reply by IceHowe** [reply]
> 用的是pgnet模型吗

没，用的PPOCRv4

**Reply by GreatV** [reply]
ppocrv4识别模型不支持这种格式，只能用pgnet。文档里也是这么写的。

**Reply by IceHowe** [reply]
> ppocrv4识别模型不支持这种格式，只能用pgnet。文档里也是这么写的。

那请问PPOCR系列支持的数据格式是什么样的啊？VOC吗？因为PPOCR里面涉及文字检测和文字识别，还是说文字检测部分用的VOC，识别部分用的图一那种文字？还是只有v4不一样，v3是一样的？

**Reply by UserWangZz** [reply]
检测用的图2的格式，但是会忽略文本内容的标签，识别用的图1的格式，文档中写的是使用pgnet训练，pgnet是一个端到端OCR的方法，所以图2的格式是可以用来训练的。

**Reply by IceHowe** [reply]
> 检测用的图2的格式，但是会忽略文本内容的标签，识别用的图1的格式，文档中写的是使用pgnet训练，pgnet是一个端到端OCR的方法，所以图2的格式是可以用来训练的。

https://github.com/PaddlePaddle/PaddleOCR/discussions/13576
我在这个讨论里写到，检测模型也有这种问题，这个格式也不能行

**Reply by IceHowe** [reply]
> ppocrv4识别模型不支持这种格式，只能用pgnet。文档里也是这么写的。

好的，谢谢
